### PR TITLE
Custom inputs, validations, progress details, progress listing and creating page to add new entries.

### PR DESCRIPTION
**Pull request by iarlen-reis**
# features
- [feat: created ButtonForm, InputForm and SelectorForm components to reuse code and styling.](https://github.com/micaelomota/live-youtube-web-app/commit/ee688ddd9dfcd1664ff95ee2809da4100ac660e7)
- [feat: Adding validations in the name, target, unity, deadline and currentValue fields on the NewTargetScreen page.](https://github.com/micaelomota/live-youtube-web-app/commit/08023dd72a0ff0f33c13794c6645c09a4cef32b5)
- [feat: creating TargetDetail page and HistoricCard component.](https://github.com/micaelomota/live-youtube-web-app/pull/3/commits/6fc3034f85fc7d6cae19cfa73b63c1456db1b55e)
- [feat: modifying layout and creating useful function to format data.](https://github.com/micaelomota/live-youtube-web-app/pull/3/commits/88eeaf989c07cfefdb3120fd28beac3b4833bc2d)
- [feat: Creating InputFormNumber component to handle number input.](https://github.com/micaelomota/live-youtube-web-app/pull/3/commits/8e5af74fbe5c41e2a747356447d54d7e2c388afa)
- [feat: creating page to add new entries.](https://github.com/micaelomota/live-youtube-web-app/pull/3/commits/d0a9c4d18aef4c02a45b244117c76699f83d300a)

# refactor
- [refactor: removing the created_At and updated_At logic from the new target page and moving it to the context.](https://github.com/micaelomota/live-youtube-web-app/commit/7ed81de8dc66c1212e17f2645f6442231d40204e)
- [refactor: Adding link around the list item to redirect to the progres](https://github.com/micaelomota/live-youtube-web-app/pull/3/commits/66f5ae7b7dc002952688822c0dece394418247a3)
- [refactor: refactoring the NewTargetScreen page.](https://github.com/micaelomota/live-youtube-web-app/pull/3/commits/d1f87f1447ecf50a28c6f6ae6fa7decef59bc6c8)
- [refactor: refactoring TargetContext and adding functionality to get a…](https://github.com/micaelomota/live-youtube-web-app/pull/3/commits/d2dbee3161b32f75a4b39181d60c6fe3c8fe9068)
- [refactor: refactoring InputForm component.](https://github.com/micaelomota/live-youtube-web-app/pull/3/commits/bc8eec510a8ef308a2c9453eb1937a688dfa4409)
- [refactor: modifying NewTargetScreen and TargetDetail pages.](https://github.com/micaelomota/live-youtube-web-app/pull/3/commits/532627e040ff566719d7e658338cbbc30253b95d)
- [refactor: refactoring TargetContext and using localstorage to store t…](https://github.com/micaelomota/live-youtube-web-app/pull/3/commits/38f6b36cfcf7ed0d15541f84b1d1a225601ca9d8)

# fixes
- [fix: remove node modules from git](https://github.com/micaelomota/live-youtube-web-app/commit/c35ec6a84e165abf74257b947ba56190b9aa69e9)

# chores
- [chore: installing and configuring tailwind](https://github.com/micaelomota/live-youtube-web-app/commit/92c63bf38f86be6f290ca91c7572b62b95c80363)

### CODE REVIEW

**Comment by micaelomota** [comment]
@iarlen-reis tem como remover os arquivos do node e fazer um rebase de main para pegar o .gitignore

**Comment by iarlen-reis** [comment]
> @iarlen-reis tem como remover os arquivos do node e fazer um rebase de main para pegar o .gitignore

> @micaelomota o .gitignore foi adicionado e a node modules foi removido no primeiro commit da PR

[fix: remove folder node modules](https://github.com/micaelomota/live-youtube-web-app/commit/c35ec6a84e165abf74257b947ba56190b9aa69e9)

**Comment by iarlen-reis** [comment]
> @iarlen-reis tem como remover os arquivos do node e fazer um rebase de main para pegar o .gitignore

> @micaelomota conflitos resolvidos.

File: frontend/src/context/TargetContext.tsx [hunk 1]
```diff
@@ -1,87 +1,75 @@
...
+  : [],);
+
+  const addTarget = useCallback((target: Omit<TargetProps, "id">) => {
+    setTargets((prev) => [...(prev || []), { ...target, id: Date.now() }]);
+  }, []);
+
+  const getTargetById = useCallback(
+    (id: number) => {
+      return targets?.find((target) => target.id === id)
+    }, [targets])
```
Comment (micaelomota) [inline frontend/src/context/TargetContext.tsx:43]: precisa fazer um rebase, a gente já está usando o firestore database

**Review by micaelomota — CHANGES_REQUESTED** [review]

**Comment by micaelomota** [comment]
Estou fechando porque fiz o rebase e mesclei no meu repo.  assiste a live https://youtube.com/live/MOu_Alj2VXI?feature=share

**Pull request closed without merging**